File: article_34/14_switch.go
// Seriál "Programovací jazyk Go"
//
// Třicátá čtvrtá část
//    Programovací jazyk Go pro skalní céčkaře
//    https://www.root.cz/clanky/programovaci-jazyk-go-pro-skalni-ceckare/
//
// Demonstrační příklad číslo 14:
//     Rozvětvení v jazyce Go.

package main

import "fmt"

func classify(x int) string {
	switch x {
	case 0:
		return "nula"
	case 2, 4, 6, 8:
		return "sudé číslo"
	case 1, 3, 5, 7, 9:
		return "liché číslo"
	default:
		return "?"
	}
}

func main() {
	for x := 0; x <= 10; x++ {
		fmt.Printf("%d: %s\n", x, classify(x))
	}
}

File: article_34/14_switch.c
/*
 Seriál "Programovací jazyk Go"

 Třicátá čtvrtá část
    Programovací jazyk Go pro skalní céčkaře
    https://www.root.cz/clanky/programovaci-jazyk-go-pro-skalni-ceckare/
*/

#include <stdio.h>

const char *classify(int x) {
	switch (x) {
	case 0:
		return "nula";
        case 2:
        case 4:
        case 6:
        case 8:
		return "sudé číslo";
	case 1:
	case 3:
	case 5:
	case 7:
	case 9:
		return "liché číslo";
	default:
		return "?";
	}
}

int main(void) {
    int x;
    for (x = 0; x <= 10; x++) {
        printf("%d: %s\n", x, classify(x));
    }
    return 0;
}
